tuareg.el

```diff
@@ -3645,35 +3682,43 @@ It is assumed that the range START-END delimit valid OCaml phrases."
 (defun tuareg-eval-region (start end)
   "Eval the current region in the OCaml REPL."
   (interactive "r")
-  (setq tuareg-interactive-last-phrase-pos-in-source start)
-  (save-excursion
-    (goto-char start)
-    (tuareg-backward-beginning-of-defun)
-    (setq start (point))
-    (setq end (cdr (tuareg-region-of-defun end)))
-    (if end
-        (tuareg-interactive--send-region start end)
-      (message "The expression after the point is not well braced."))))
+  (tuareg-interactive--send-region start end))
 
 (define-obsolete-function-alias 'tuareg-narrow-to-phrase #'narrow-to-defun
   "Apr 10, 2019")
 
 (defun tuareg-eval-phrase ()
-  "Eval the surrounding OCaml phrase (or block) in the OCaml REPL."
+  "Eval the surrounding OCaml phrase (or block) in the OCaml REPL.
+If the region is active, evaluate all phrases intersecting the region."
   (interactive)
-  (let ((opoint (point)))
-    ;; Move before the comment, if we are in one.
-    (let ((ppss (syntax-ppss)))
-      (if (nth 4 ppss) (goto-char (- (nth 8 ppss) 1))))
-    (let ((phrase (tuareg-discover-phrase)))
-      (if phrase
-          (progn
-            (tuareg-interactive--send-region (car phrase) (cadr phrase))
-            (if tuareg-skip-after-eval-phrase
-              (progn (goto-char (car (cddr phrase)))
-                     (tuareg-skip-blank-and-comments))
-              (goto-char opoint)))
-        (message "The expression after the point is not well braced.")))))
+  (let ((opoint (point))
+        start end)
+    (cond
+     ((region-active-p)
+      (setq start (min (point) (mark)))
+      (setq end (max (point) (mark)))
+      ;; Extend the region at the endpoints if they are in a phrase.
+      (dolist (pos (list start end))
```

Comment:
@monnier told me by email: There's also `region-beginning` and `region-end` if you want them in-order.

Reply:
`region-beginning` and `region-end` are clearly what I should use, thank you!. I don't think we can initialise both `start` and `end` with `(point)` – if we can, the logic is not obvious and I prefer not too.

I really should have known better since I'm a member of the Society for Emacs Region Reform! (If you join, I'll waive the membership dues...)

Reply:
I'm fine with `region-beginning` and `region-end`.

Reply:
Good, I had to rewrite this section anyway because of the bug you found below.